### PR TITLE
DRIVERS-2878 OIDC Atlas Testing Updates

### PR DESCRIPTION
**Pull request by blink1073**
Please complete the following before merging:

- [x] Update changelog.
- [ ] Make sure there are generated JSON files from the YAML test files.
- [x] Test changes in at least one language driver.   https://github.com/mongodb/mongo-python-driver/pull/1567
- [ ] Test these changes against all server versions and topologies (including standalone, replica set, sharded
  clusters, and serverless).

### CODE REVIEW

File: source/auth/tests/mongodb-oidc.md [hunk 1]
```diff
@@ -36,25 +25,25 @@ Drivers MUST run the prose tests in all supported OIDC environments.
...
-fetched from the `drivers/oidc` Secrets vault, see
-[vault instructions](https://wiki.corp.mongodb.com/display/DRIVERS/Using+AWS+Secrets+Manager+to+Store+Testing+Secrets).
-Use `OIDC_ATLAS_URI_SINGLE` for the `MONGODB_URI`. If using local servers is preferred, using the
-[Local Testing](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/auth_oidc/README.md#local-testing)
-method, use `mongodb://localhost/?authMechanism=MONGODB-OIDC` for `MONGODB_URI`.
+After setting up your OIDC
+[environment](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/auth_oidc/README.md),
+source the `secrets-export.sh` file and use the associated env variables in your tests.
+
+The default client should set the appropriate `ENVIRONMENT` auth mechanism property and include a callback that gets the
```
Comment (katcharov) [inline source/auth/tests/mongodb-oidc.md:32]:
```suggestion
An OIDC configured client MUST set the appropriate `ENVIRONMENT` auth mechanism property and include a callback that gets the
```
To clarify the term "OIDC configured client", which is used later?

File: source/auth/tests/mongodb-oidc.md [hunk 2]
```diff
@@ -152,71 +141,58 @@ ______________________________________________________________________
...
+`test_user1@${OIDC_DOMAIN}`.
 
 ### (1) OIDC Human Callback Authentication
 
 Drivers MUST be able to authenticate using OIDC callback(s) when there is one principal configured.
 
 **1.1 Single Principal Implicit Username**
 
-- Create default OIDC client with `authMechanism=MONGODB-OIDC`.
+- Create default client
```
Comment (katcharov) [inline source/auth/tests/mongodb-oidc.md:164]: Is it correct that a "default client" is the same as an "OIDC configured client"? If so, suggest using the more specific "OIDC configured client" term throughout, including the "create a client" statements below.

**Review by katcharov — CHANGES_REQUESTED** [review]
Looks good, just minor clarifications

File: source/auth/tests/mongodb-oidc.md (hunk 2 above)
Comment (blink1073) [reply source/auth/tests/mongodb-oidc.md]: Updated for consistency.

**Review by durran — APPROVED** [review]

**Review by katcharov — APPROVED** [review]
LGTM!

**Pull request merged**